reactor-core/src/main/java/reactor/core/publisher/FluxSwitchOnFirst.java

```diff
@@ -471,27 +480,26 @@ public final void onError(Throwable t) {
 
 			final long previousState = markInboundTerminated(this);
 			if (hasInboundCancelled(previousState) || hasInboundTerminated(previousState)) {
-				Operators.onErrorDropped(t, this.outer.currentContext());
+				Operators.onErrorDropped(t, this.outboundSubscriber.currentContext());
 				return;
 			}
 
 			if (hasFirstValueSent(previousState)) {
 				synchronized (this) {
-					this.inner.onError(t);
+					this.inboundSubscriber.onError(t);
 				}
 				return;
 			}
 
 			if (!hasFirstValueReceived(previousState)) {
 				final Publisher<? extends R> result;
-				final CoreSubscriber<? super R> o = this.outer;
+				final CoreSubscriber<? super R> o = this.outboundSubscriber;
 				try {
-					result = Objects.requireNonNull(this.transformer.apply(Signal.error(t,
-							o.currentContext()), this),
-							"The transformer returned a null value");
+					final Signal<T> signal = Signal.error(t, o.currentContext());
+					result = Objects.requireNonNull(this.transformer.apply(signal, this), "The transformer returned a null value");
 				}
 				catch (Throwable e) {
-					o.onError(Exceptions.addSuppressed(e, t));
+					o.onError(Exceptions.addSuppressed(t, e));
```

Comment:
I don't remember if this was tested in unit tests before. shouldn't be too hard to add a test for this (validate the outer throwable is suppressing the function's exception

Reply:
done. unit tests are in place for that